ttnn/cpp/ttnn/operations/moreh/moreh_adam/moreh_adam.hpp

```diff
@@ -0,0 +1,37 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/operations/moreh/moreh_adam/device/moreh_adam_device_operation.hpp"
+
+namespace ttnn::operations::moreh::moreh_adam {
+struct MorehAdam {
+    static std::vector<std::optional<Tensor>> invoke(
+        const Tensor& param_in,
+        const Tensor& grad,
+        const Tensor& exp_avg_in,
+        const Tensor& exp_avg_sq_in,
+        float lr,
+        float beta1,
+        float beta2,
+        float eps,
+        float weight_decay,
+        uint32_t step,
+        bool amsgrad,
+        const std::optional<const Tensor> max_exp_avg_sq_in,
+        const std::optional<const Tensor> param_out,
+        const std::optional<const Tensor> exp_avg_out,
+        const std::optional<const Tensor> exp_avg_sq_out,
+        const std::optional<const Tensor> max_exp_avg_sq_out,
+        const std::optional<ttnn::MemoryConfig>& memory_config,
+        const DeviceComputeKernelConfig compute_kernel_config);
+};
+}  // namespace ttnn::operations::moreh::moreh_adam
+
+namespace ttnn {
+constexpr auto moreh_adam =
+    ttnn::register_operation<"ttnn::moreh_adam", ttnn::operations::moreh::moreh_adam::MorehAdam>();
```

Comment:
```suggestion
    ttnn::register_operation_with_auto_launch_op<"ttnn::moreh_adam", ttnn::operations::moreh::moreh_adam::MorehAdam>();
```
There seems to be an issue related to the queue in register_operation at the moment. Please use register_operation_with_auto_launch_op instead

Reply:
@hschoi4448 Isn’t the note that was sent to us says to only do so to primitives? And `ttnn:moreh_adam` is a composite operation